8vim/src/main/java/inc/flide/vim8/views/mainKeyboard/XpadView.java

```diff
@@ -222,18 +230,18 @@ private void paintTypingTrail(Canvas canvas) {
                             Color.TRANSPARENT,
                             Shader.TileMode.CLAMP));
 
+                    SharedPreferences sp_opacity_value = getContext().getSharedPreferences(getContext().getString(R.string.basic_preference_file_name), Activity.MODE_PRIVATE);
+                    int current_opacity_value = sp_opacity_value.getInt(getContext().getString(R.string.storing_opacity_value_in_sharedPreference),0);
+
+                    typingTrailPaint.setAlpha(current_opacity_value);
                     canvas.drawCircle(x, y, trailRadius, typingTrailPaint);
                 }
             }
         }
         canvas.drawPath(typingTrailPath, typingTrailPaint);
     }
 
-    private int getTrailColor() {
-        SharedPreferences sharedPreferences = this.getContext().getSharedPreferences(this.getContext().getString(R.string.basic_preference_file_name), Activity.MODE_PRIVATE);
-        int trailColor = sharedPreferences.getInt(this.getContext().getString(R.string.color_selection), Color.YELLOW);
-        return trailColor;
```

Comment:
I don't understand, When there existed a separate function to figure out the color for the trail why would you not reuse it?
Why did you delete this and put the code back into the previous function? Does this make the code more readable? reusable?
Fix this.